keras/engine/training_test.py

```diff
@@ -2390,6 +2390,63 @@ def call(
         self.assertEqual(input_specs[1].shape.as_list(), [2, 2])
         self.assertEqual(input_specs[2].shape.as_list(), [3, 3])
 
+    def test_states(self):
+        inp = keras.Input(batch_shape=[1, 2, 3])
+
+        stateless = keras.layers.SimpleRNN(units=3, stateful=False)
+        stateful = keras.layers.SimpleRNN(units=3, stateful=True)
+
+        model = keras.Model(inp, [stateless(inp), stateful(inp)])
+
+        states = model.states
+        states_ = stateful.states
```

Comment:
For naming convention, we should change those 2 var to model_states and layer_states.